src/validation.h

```diff
@@ -196,22 +196,33 @@ static const uint64_t MIN_DISK_SPACE_FOR_BLOCK_FILES = 550 * 1024 * 1024;
  * block is made active. Note that it does not, however, guarantee that the
  * specific block passed to it has been checked for validity!
  *
- * If you want to *possibly* get feedback on whether pblock is valid, you must
- * install a CValidationInterface (see validationinterface.h) - this will have
- * its BlockChecked method called whenever *any* block completes validation.
+ * Performs initial sanity checks using the provided CValidationState before
+ * connecting any block(s). If you want to *possibly* get feedback on whether
+ * pblock is valid beyond just cursory mutation/DoS checks, you must install
+ * a CValidationInterface (see validationinterface.h) - this will have its
```

Comment:
"The installed `BlockChecked` method will be called for *any* block completing validation. It may not be called for the provided pblock as this one not being part of the best chain. But if `BlockChecked` is called for it and its `CValidationState` is invalid, the invalidity reason will be different than the `CValidationState got back from `ProcessNewBlock`". Just a verbose attempt to underscores clearly the fact you will get 2 different set of invalidity reasons.